lib/pwa.js

```diff
@@ -150,23 +151,27 @@ function updateMetadataDescription(pwa) {
   });
 }
 
-function updateIcon(pwa, manifest) {
+exports.updateIcon = function(pwa, manifest) {
   const url = manifest.getBestIconUrl();
-  const extension = url.substring(url.lastIndexOf('.'), url.length);
+  const extension = url.substring(url.lastIndexOf('.'), url.length).split('?')[0];
```

Comment:
This can also be done using node modules via something like `path.extname(url.parse(manifest.getBestIconUrl()).pathname)`. (Slightly more semantic though `path` and `url` need to be imported.)